### ISSUE TITLE
Unexpected PrivateIdentifier in name expression with literal-like access

### ISSUE THREAD

**Issue by sviridoff** [microsoft/TypeScript#48662]
# Bug Report

### 🔎 Search Terms

"Unexpected PrivateIdentifier in name expression with literal-like access."
"Private Identifier"
"isSameEntityName"
"isPrivateIdentifier"

### 🕗 Version & Regression Information

TypeScript version 4.6.2

### ⏯ Playground Link

[Playground Link](https://www.typescriptlang.org/play?#code/MYGwhgzhAECC0G8BQ1XQMQAcwCcwFsIBuJFNCAF1woAoBKRMtNYAewDtLps9DoBeaBQAWASwgA6LLgIwAPnMQBfEk2bQ2nViACmEkKwDmNHrLolmSpFaSauYAdHY6A7nHqqwEytQ9IgA)

### 💻 Code

```javascript
class A {
    #params;

    start() {
        const params = this.#params || {}; // Here the issue

        console.log(params);
    }
}

const a = new A();

a.start();
```

### 🙁 Actual behavior

It throws unexpected error on compilation

### 🙂 Expected behavior

No errors expected

**Comment by MartinJohns** [comment]
Duplicate of #48487.
